src/test/java/org/openRealmOfStars/player/combat/CombatTest.java

```diff
@@ -1,5 +1,524 @@
+	}
+	@Test
+	/**
+	 * Purpose: When there are only dependerShip, to make sure that handleEndCombat works well
+	 * Input: handleEndCombat(). there is only dependerShip.
+	 * Expected: Verify that the methods that make up handleEndCombat() are working.
+	 * 			attackerFleet.setPos is never called.
+	 * 			attackerFleetList.getIndexByName is called once.
+	 * 			attackerFleetList.remove is called once.
+	 */
+	public void testDependerWinEndCombat(){
+		attackerShips = new Ship[]{};
+		dependerShips = new Ship[]{dependerShip};
+		when(attackerFleet.getShips()).thenReturn(attackerShips);
+		when(dependerFleet.getShips()).thenReturn(dependerShips);
+		Combat combatTestItem = new Combat(attackerFleet, dependerFleet, attackerInfo, dependerInfo);
+		Coordinate dependerFleetCoordinate = mock(Coordinate.class);
+		when(dependerFleet.getCoordinate()).thenReturn(dependerFleetCoordinate);
+		doNothing().when(attackerFleet).setPos(dependerFleetCoordinate);
+		when(dependerFleetList.getIndexByName(anyString())).thenReturn(1);
+		when(attackerFleetList.getIndexByName(anyString())).thenReturn(1);
+		combatTestItem.isCombatOver();
+		combatTestItem.handleEndCombat();
+		verify(attackerFleet,never()).setPos(dependerFleetCoordinate);;
+		verify(attackerFleetList,times(1)).getIndexByName(anyString());
+		verify(attackerFleetList,times(1)).remove(anyInt());
+	}
+	@Test
+	/**
+	 * Purpose: When there are two ships, to make sure that handleEndCombat works well
+	 * Input: handleEndCombat(). there are two Ships.
+	 * Expected: Verify that the methods that make up handleEndCombat() are working.
+	 * 			attackerFleet.setPos is never called.
+	 * 			attackerFleetList.getIndexByName is never called.
+	 * 			attackerFleetList.remove is never called.
+	 * 			dependerFleetList.getIndexByName is never called.
+	 * 			dependerFleetList.remove is never called.
+	 */
+	public void testNoWinerEndCombat(){
+		attackerShips = new Ship[]{attackerShip};
+		dependerShips = new Ship[]{dependerShip};
+		when(attackerFleet.getShips()).thenReturn(attackerShips);
+		when(dependerFleet.getShips()).thenReturn(dependerShips);
+		Combat combatTestItem = new Combat(attackerFleet, dependerFleet, attackerInfo, dependerInfo);
+		Coordinate dependerFleetCoordinate = mock(Coordinate.class);
+		when(dependerFleet.getCoordinate()).thenReturn(dependerFleetCoordinate);
+		doNothing().when(attackerFleet).setPos(dependerFleetCoordinate);
+		when(dependerFleetList.getIndexByName(anyString())).thenReturn(1);
+		when(attackerFleetList.getIndexByName(anyString())).thenReturn(1);
+		combatTestItem.isCombatOver();
+		combatTestItem.handleEndCombat();
+		
+		verify(attackerFleet,never()).setPos(dependerFleetCoordinate);;
+		verify(dependerFleetList,never()).getIndexByName(anyString());
+		verify(dependerFleetList,never()).remove(anyInt());
+		verify(attackerFleetList,never()).getIndexByName(anyString());
+		verify(attackerFleetList,never()).remove(anyInt());
+	}
+=======
```

Comment:
After this there are some leftovers which should be at the very beginning of this file.